src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateImplicitNamespaceImports.targets

```diff
@@ -0,0 +1,57 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.GenerateImplicitNamespaceImports.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--
+    ============================================================
+                                     GenerateImplicitNamespaceImports
+
+    Generates implicit namespace imports source to intermediate directory for C# projects
+    ============================================================
+    -->
+  <PropertyGroup>
+    <GeneratedImplicitNamespaceImportFile Condition="'$(GeneratedImplicitNamespaceImportFile)' ==''">$(IntermediateOutputPath)$(MSBuildProjectName).ImplicitNamespaceImports$(DefaultLanguageSourceExtension)</GeneratedImplicitNamespaceImportFile>
+    <DisableImplicitNamespaceImports_DotNet Condition="'$(DisableImplicitNamespaceImports_DotNet)' == '' and '$(DisableImplicitNamespaceImports)' != 'true'">false</DisableImplicitNamespaceImports_DotNet>
+  </PropertyGroup>
+
+  <!--
+    Note that this must run before every invocation of CoreCompile to ensure that all compiler
+    runs see the generated assembly info. There is at least one scenario involving Xaml
+    where CoreCompile is invoked without other potential hooks such as Compile or CoreBuild,
+    etc., so we hook directly on to CoreCompile. Furthermore, we  must run *after*
+    PrepareForBuild to ensure that the intermediate directory has been created.
+
+    Targets that generate Compile items are also expected to run before
+    BeforeCompile targets (common targets convention).
+   -->
+  <Target Name="GenerateImplicitNamespaceImports"
+          BeforeTargets="BeforeCompile;CoreCompile"
+          DependsOnTargets="PrepareForBuild;CoreGenerateImplicitNamespaceImports"
+          Condition="'$(DisableImplicitNamespaceImports)' != 'true'" />
+
+  <Target Name="CoreGenerateImplicitNamespaceImports"
+          Outputs="$(GeneratedImplicitNamespaceImportFile)"
+          Condition="'@(Import)' != ''">
+    <ItemGroup>
+      <_UniqueImport Include="@(Import->Distinct())" />
+      <_ImportFileLine Include="// %3Cautogenerated />"/>
+      <_ImportFileLine Include="global using global::%(_UniqueImport.Identity)%3B"/>
```

Comment:
Should we have the `global::` appear in the itemgroup? This thing would prevent something like

`<Import Include="static Microsoft.AspNetCore.Http.Results" />` from working

Reply:
No global shouldn't appear in the item group this feature is now aligned with the VB import feature.